### PR TITLE
📝 Document API to handle access requests to gated repositories

### PR DESCRIPTION
**Pull request by SBrandeis**
To be merged after https://github.com/huggingface/moon-landing/pull/6821

### CODE REVIEW

**Review by julien-c — COMMENTED** [review]
looks good to me!

**Review by Michellehbn — APPROVED** [review]
ty!

File: docs/hub/datasets-gated.md [hunk 1]
```diff
@@ -38,6 +38,18 @@ Only users whose access requests have been approved will be able to access the d
...
 </div>
 
+You can automate the approval of access requests with the following API:
+
+| Method | URI | Description | Payload |
+| ------ | --- | ----------- | -------  |
+| `GET` | `/api/datasets/{repo_id}/user-access-request/pending` | Retrieve the list of pending access requests for the given model | ```headers = { "authorization" :  "Bearer $token" }``` |
+| `GET` | `/api/datasets/{repo_id}/user-access-request/accepted` | Retrieve the list of accepted access requests for the given model | ```headers = { "authorization" :  "Bearer $token" }``` |
+| `GET` | `/api/datasets/{repo_id}/user-access-request/rejected` | Retrieve the list of rejected access requests for the given model | ```headers = { "authorization" :  "Bearer $token" }``` |
+| `POST` | `/api/datasets/{repo_id}/user-access-request/handle` | Change the status of a given access request from `expectedStatus` to `status`. `expectedStatus` is optional, and defaults to `"pending"`. | ```headers = { "authorization" :  "Bearer $token" }``` ```json = { "status": "accepted" | "rejected" | "pending", "userId": "abcdef1234", "expectedStatus": "pending" | "accepted" | "rejected" } ``` |
```
Comment (osanseviero) [inline docs/hub/datasets-gated.md:48]:
```suggestion
| `GET` | `/api/datasets/{repo_id}/user-access-request/pending` | Retrieve the list of pending access requests for the given dataset. | ```headers = { "authorization" :  "Bearer $token" }``` |
| `GET` | `/api/datasets/{repo_id}/user-access-request/accepted` | Retrieve the list of accepted access requests for the given dataset. | ```headers = { "authorization" :  "Bearer $token" }``` |
| `GET` | `/api/datasets/{repo_id}/user-access-request/rejected` | Retrieve the list of rejected access requests for the given dataset. | ```headers = { "authorization" :  "Bearer $token" }``` |
| `POST` | `/api/datasets/{repo_id}/user-access-request/handle` | Change the status of a given access request from `expectedStatus` to `status`. `expectedStatus` is optional, and defaults to `"pending"`. | ```headers = { "authorization" :  "Bearer $token" }``` ```json = { "status": "accepted" | "rejected" | "pending", "userId": "abcdef1234", "expectedStatus": "pending" | "accepted" | "rejected" } ``` |
```

**Review by osanseviero — APPROVED** [review]
LGTM! Just a couple of minor nits. THanks!

**Pull request merged**